portfolio/src/main/webapp/script.js

```diff
@@ -39,21 +39,20 @@ function randQuote() {
 // Fetch comments from DataServlet.java
 function getComments() {
 
-  fetch('/comments').then(response => response.json()).then((comments) => {
+  fetch('/comments').then(response => response.json()).then((cmtContainer) => {
 
-    var str_comments = "<h2>Comments</h2><p>";
+    if(!cmtContainer) return;
 
-    for (var index in comments){
-      str_comments += comments[index] + "<br>";
+    var str_comments = "<h2>Comments</h2>";
+
+    for (var index in cmtContainer) {
+      str_comments += ("<li>" + cmtContainer[index] + "</li>");
     }
-    
-    str_comments.concat("</p>");
 
-    document.getElementById("comments_sec").innerHTML = str_comments;
+    document.getElementById("comments-output").innerHTML = str_comments;
```

Comment:
Avoid using `innerHTML` as it opens you up to [XSS vulnerabilities](https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html). I think gson might sanitize this for you but it's still good practice to take precautions on the front-end. `innerText` should be safe here.

Reply:
Should I replace .innerHTML with document.write?

Reply:
> Should I replace .innerHTML with document.write?

Never mind, I will use the DOM API.